### PR TITLE
Trial3

### PR DESCRIPTION
**Pull request by manimanjari**
1.removed all the txt files
2.used functions to reduce redundancy
3.taking path as an argument
4.tried to indent it

### CODE REVIEW

File: try2.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	do
+		get_page_link "$movieName";
+		imdbRating=$(curl -s "http://www.imdb.com$writeLinksFileName"  \
+			| sed -e '/itemprop="ratingValue"/,/findMoreMatches/!d'  \
+			| awk '{for(i=1;i<=NF;i++) if ($i=="based")  print $(i-1)}'  \
+			| cut -c8-10);
+		echo -n  "${movieName} ";
+		echo "${imdbRating}";
+	done;	
+}
```
Comment (vishesh92) [inline try2.sh]: Trailing white spaces.

File: try2.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    get_imdb_rating "$1" | sort -nrk2 | sed 's/+/ /g';
```
Comment (vaidik) [inline try2.sh]: White spaces are not required here.

File: try2.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	for searchText in $1/*
+	do
+		get_page_link "$movieName";
+		imdbRating=$(curl -s "http://www.imdb.com$writeLinksFileName"  \
+			| sed -e '/itemprop="ratingValue"/,/findMoreMatches/!d'  \
+			| awk '{for(i=1;i<=NF;i++) if ($i=="based")  print $(i-1)}'  \
+			| cut -c8-10);
+		echo -n  "${movieName} ";
+		echo "${imdbRating}";
+	done;
```
Comment (vishesh92) [inline try2.sh]: Trailing white spaces.

File: try2.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	movieName=${movieName// /+};
+	movieName=${movieName//./+};
+	movieName=${movieName//_/+};
+}
+
+# This function enters the movie name in the search url and extracts 
+# the link for first suggestion on the page.
+get_page_link() {
+	get_name "$searchText";
+	writeLinksFileName=$(curl -s "http://www.imdb.com/find?q=$movieName"  \
```
Comment (vishesh92) [inline try2.sh]: use only one space before `\`.

Comment (vaidik) [inline try2.sh]: You are using global variables here. The variable created in another function should not be used in this function. You should return the movieName from the other function `get_name` by doing this `echo $movieName` in the end and then capture that value in the previous line and use it in this function.

File: try2.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+		| sed -e '/Titles<\/h3>/,/findMoreMatches/!d'  \
+		| grep -E -w -o "\/title\/[a-zA-Z0-9]+\/" | head -1)
+}
+
+# This function takes names from the directory and extracts imdb rating
+get_imdb_rating() {
+	for searchText in $1/*
+	do
+		get_page_link "$movieName";
```
Comment (vaidik) [inline try2.sh]: Same as issue as mentioned before.

Comment (vaidik) [inline try2.sh]: Using global scope is evil

**Pull request merged**